passing tests and name changes

LukasKlockenhoff opened the pull request.

jonasliendl commented on casebased/components/similarity_measure/similarity.py:
```diff
@@ -13,12 +13,23 @@
-        self.vocabulary = vocabulary
+    def __init__(
+        self,
+        k: Optional[int] = None,
+        k_optimizer: Optional[KAlgorithm] = None,
+        similarity_measure: Optional[
+            Union[SimilarityMeasureAlgorithm, Callable[[list, list], int]]
```
Hier kannst du auch einfach ein Configuration Objekt erwarten. Das hat alle drei Sachen drinnen

jonasliendl replied:
Optional

LukasKlockenhoff replied:
done

jonasliendl commented on casebased/components/similarity_measure/similarity.py:
```diff
@@ -13,12 +13,23 @@
+    ):
+        # self.case_base = case_base
+        # self.vocabulary = vocabulary
+        k = k
+        k_optimizer = k_optimizer
+        similarity_measure = similarity_measure
```
```suggestion
        self.k = k
        self.k_optimizer = k_optimizer
        self.similarity_measure = similarity_measure
```

jonasliendl commented on casebased/components/similarity_measure/similarity.py:
```diff
@@ -6,19 +6,24 @@
-        self.case_base = case_base
-        self.vocabulary = vocabulary
+    def __init__(self, config: Configuration):
+        # self.case_base = case_base
+        # self.vocabulary = vocabulary
```
```suggestion
```

jonasliendl commented on casebased/components/similarity_measure/similarity.py:
```diff
@@ -28,14 +33,16 @@ def fit(self, **kwargs):
-    def _fit_classifier(self, **kwargs):
-        k = kwargs.get("k")
+    def _fit_classifier(self, case_base: CaseBase, vocabulary: Vocabulary, **kwargs):
+        k = kwargs.get("k") if kwargs.get("k") else self.k
```
```suggestion
        k = kwargs.get("k") or self.k
```

jonasliendl commented on casebased/components/similarity_measure/similarity.py:
```diff
@@ -6,19 +6,24 @@
+    def __init__(self, config: Configuration):
+        # self.case_base = case_base
+        # self.vocabulary = vocabulary
+        self.k = config.k
```
```suggestion
        self.k = config.k or "auto"
```

jonasliendl commented on casebased/components/similarity_measure/similarity.py:
```diff
@@ -28,14 +33,16 @@ def fit(self, **kwargs):
-        k = kwargs.get("k")
+    def _fit_classifier(self, case_base: CaseBase, vocabulary: Vocabulary, **kwargs):
+        k = kwargs.get("k") if kwargs.get("k") else self.k
+        if k is None:
+            k = "auto"
```
```suggestion
```
Kann gelöscht werden wenn k by default "auto" ist. Siehe anderes Kommentar in Review. Bei k hatte ich mir auch vorgestellt, dass das k für ein System gespeichert wird und nicht jedes mal neu berechnet werden muss. Das ist aber Scope von einem anderen Ticket.

jonasliendl reviewed (changes requested):
Danach wird das direkt Approved.

LukasKlockenhoff commented:
This pull request includes several changes to improve the consistency and functionality of the case-based reasoning system. The main changes involve renaming variables for clarity, updating method signatures, and modifying the configuration and initialization processes.

### Changes to CaseBase and Case Management:

* [`casebased/components/casebase/casebase.py`](diffhunk://#diff-9bcedba2717c2a63c73e773d5ee7347f042dea3213df8eed92df122f544045f2L14-R33): Renamed the `df` variable to `cases` throughout the `CaseBase` class to improve clarity.
* [`casebased/actors/adapter.py`](diffhunk://#diff-08011ade3b058c72896627d85a2ce46f9526ff3b94ba3aa59da69e2731a0c387L14-R14): Updated the `adapt` method to use `cases` instead of `data` for consistency with the new `CaseBase` variable names.

### Changes to Similarity Measure:

* [`casebased/components/similarity_measure/similarity.py`](diffhunk://#diff-bb65f5f3707feb44e5f9dfd29f3c7e7a8be7d846fde0d038c02cc2c59d90682bR9-R24): Updated the `SimilarityMeasure` class to accept a `Configuration` object in its constructor and modified related methods to use this configuration. [[1]](diffhunk://#diff-bb65f5f3707feb44e5f9dfd29f3c7e7a8be7d846fde0d038c02cc2c59d90682bR9-R24) [[2]](diffhunk://#diff-bb65f5f3707feb44e5f9dfd29f3c7e7a8be7d846fde0d038c02cc2c59d90682bL31-R41) [[3]](diffhunk://#diff-bb65f5f3707feb44e5f9dfd29f3c7e7a8be7d846fde0d038c02cc2c59d90682bR68-R69) [[4]](diffhunk://#diff-bb65f5f3707feb44e5f9dfd29f3c7e7a8be7d846fde0d038c02cc2c59d90682bL73-R87)

### Changes to Vocabulary:

* [`casebased/components/vocabulary/attribute.py`](diffhunk://#diff-e44595c3656920cd6ac592078bde50cab9dfac90e24f6bcc8ed74e0ede3ac759L149-R149): Corrected the `data_type` parameter in `TargetAttribute` and `FeatureAttribute` to use lists instead of tuples. [[1]](diffhunk://#diff-e44595c3656920cd6ac592078bde50cab9dfac90e24f6bcc8ed74e0ede3ac759L149-R149) [[2]](diffhunk://#diff-e44595c3656920cd6ac592078bde50cab9dfac90e24f6bcc8ed74e0ede3ac759L165-R165)
* [`casebased/components/vocabulary/vocabulary.py`](diffhunk://#diff-6574c9fa7f4c741a4ba047e4ba6ffedc7a79602e6642ebcd5bbff1377c9a89c8L24-R40): Modified the `Vocabulary` class to handle `None` values for `features` and `targets`, ensuring default empty lists are used. [[1]](diffhunk://#diff-6574c9fa7f4c741a4ba047e4ba6ffedc7a79602e6642ebcd5bbff1377c9a89c8L24-R40) [[2]](diffhunk://#diff-6574c9fa7f4c741a4ba047e4ba6ffedc7a79602e6642ebcd5bbff1377c9a89c8L107-R126)

### Changes to Configuration:

* [`casebased/config.py`](diffhunk://#diff-604c24b9a40d0fb49a4836b5dd13b0d7374e5dbc6b1a2862755e31305376d71bL37-R37): Renamed `k_algorithm` to `k_optimizer` in the `Configuration` class to better reflect its purpose.

### Changes to Tests:

* [`tests/test_case_base_system.py`](diffhunk://#diff-db32e78d1db3dd33b209e2e7a936aac92f7b912e5b78594badf92175176b3feeL11-R17): Updated test configurations to use the new `Configuration` class and variable names.
* [`tests/test_similarity_measure.py`](diffhunk://#diff-8695ddc9c0b885aa245ef2d881fdae33ca029cae6e66a0c2985d80aee98a40b0R5-R12): Adjusted tests to reflect changes in the `SimilarityMeasure` class and `CaseBase` initialization. [[1]](diffhunk://#diff-8695ddc9c0b885aa245ef2d881fdae33ca029cae6e66a0c2985d80aee98a40b0R5-R12) [[2]](diffhunk://#diff-8695ddc9c0b885aa245ef2d881fdae33ca029cae6e66a0c2985d80aee98a40b0L18-R22) [[3]](diffhunk://#diff-8695ddc9c0b885aa245ef2d881fdae33ca029cae6e66a0c2985d80aee98a40b0L33-R38) [[4]](diffhunk://#diff-8695ddc9c0b885aa245ef2d881fdae33ca029cae6e66a0c2985d80aee98a40b0R48-R52)
* [`tests/test_vocabulary.py`](diffhunk://#diff-46986df913746988c72be5388d480278d6a98ea3cd44b216ab8d2b4431ad0cd0L15-R15): Fixed the `data_type` parameter in `TargetAttribute` initialization.

jonasliendl commented:
> This pull request includes several changes to improve the consistency and functionality of the case-based reasoning system. The main changes involve renaming variables for clarity, updating method signatures, and modifying the configuration and initialization processes.
> 
> ### Changes to CaseBase and Case Management:
> * [`casebased/components/casebase/casebase.py`](https://github.com/Case-Based/casebased/pull/19/files#diff-9bcedba2717c2a63c73e773d5ee7347f042dea3213df8eed92df122f544045f2L14-R33): Renamed the `df` variable to `cases` throughout the `CaseBase` class to improve clarity.
> * [`casebased/actors/adapter.py`](https://github.com/Case-Based/casebased/pull/19/files#diff-08011ade3b058c72896627d85a2ce46f9526ff3b94ba3aa59da69e2731a0c387L14-R14): Updated the `adapt` method to use `cases` instead of `data` for consistency with the new `CaseBase` variable names.
> 
> ### Changes to Similarity Measure:
> * [`casebased/components/similarity_measure/similarity.py`](https://github.com/Case-Based/casebased/pull/19/files#diff-bb65f5f3707feb44e5f9dfd29f3c7e7a8be7d846fde0d038c02cc2c59d90682bR9-R24): Updated the `SimilarityMeasure` class to accept a `Configuration` object in its constructor and modified related methods to use this configuration. [[1]](https://github.com/Case-Based/casebased/pull/19/files#diff-bb65f5f3707feb44e5f9dfd29f3c7e7a8be7d846fde0d038c02cc2c59d90682bR9-R24) [[2]](https://github.com/Case-Based/casebased/pull/19/files#diff-bb65f5f3707feb44e5f9dfd29f3c7e7a8be7d846fde0d038c02cc2c59d90682bL31-R41) [[3]](https://github.com/Case-Based/casebased/pull/19/files#diff-bb65f5f3707feb44e5f9dfd29f3c7e7a8be7d846fde0d038c02cc2c59d90682bR68-R69) [[4]](https://github.com/Case-Based/casebased/pull/19/files#diff-bb65f5f3707feb44e5f9dfd29f3c7e7a8be7d846fde0d038c02cc2c59d90682bL73-R87)
> 
> ### Changes to Vocabulary:
> * [`casebased/components/vocabulary/attribute.py`](https://github.com/Case-Based/casebased/pull/19/files#diff-e44595c3656920cd6ac592078bde50cab9dfac90e24f6bcc8ed74e0ede3ac759L149-R149): Corrected the `data_type` parameter in `TargetAttribute` and `FeatureAttribute` to use lists instead of tuples. [[1]](https://github.com/Case-Based/casebased/pull/19/files#diff-e44595c3656920cd6ac592078bde50cab9dfac90e24f6bcc8ed74e0ede3ac759L149-R149) [[2]](https://github.com/Case-Based/casebased/pull/19/files#diff-e44595c3656920cd6ac592078bde50cab9dfac90e24f6bcc8ed74e0ede3ac759L165-R165)
> * [`casebased/components/vocabulary/vocabulary.py`](https://github.com/Case-Based/casebased/pull/19/files#diff-6574c9fa7f4c741a4ba047e4ba6ffedc7a79602e6642ebcd5bbff1377c9a89c8L24-R40): Modified the `Vocabulary` class to handle `None` values for `features` and `targets`, ensuring default empty lists are used. [[1]](https://github.com/Case-Based/casebased/pull/19/files#diff-6574c9fa7f4c741a4ba047e4ba6ffedc7a79602e6642ebcd5bbff1377c9a89c8L24-R40) [[2]](https://github.com/Case-Based/casebased/pull/19/files#diff-6574c9fa7f4c741a4ba047e4ba6ffedc7a79602e6642ebcd5bbff1377c9a89c8L107-R126)
> 
> ### Changes to Configuration:
> * [`casebased/config.py`](https://github.com/Case-Based/casebased/pull/19/files#diff-604c24b9a40d0fb49a4836b5dd13b0d7374e5dbc6b1a2862755e31305376d71bL37-R37): Renamed `k_algorithm` to `k_optimizer` in the `Configuration` class to better reflect its purpose.
> 
> ### Changes to Tests:
> * [`tests/test_case_base_system.py`](https://github.com/Case-Based/casebased/pull/19/files#diff-db32e78d1db3dd33b209e2e7a936aac92f7b912e5b78594badf92175176b3feeL11-R17): Updated test configurations to use the new `Configuration` class and variable names.
> * [`tests/test_similarity_measure.py`](https://github.com/Case-Based/casebased/pull/19/files#diff-8695ddc9c0b885aa245ef2d881fdae33ca029cae6e66a0c2985d80aee98a40b0R5-R12): Adjusted tests to reflect changes in the `SimilarityMeasure` class and `CaseBase` initialization. [[1]](https://github.com/Case-Based/casebased/pull/19/files#diff-8695ddc9c0b885aa245ef2d881fdae33ca029cae6e66a0c2985d80aee98a40b0R5-R12) [[2]](https://github.com/Case-Based/casebased/pull/19/files#diff-8695ddc9c0b885aa245ef2d881fdae33ca029cae6e66a0c2985d80aee98a40b0L18-R22) [[3]](https://github.com/Case-Based/casebased/pull/19/files#diff-8695ddc9c0b885aa245ef2d881fdae33ca029cae6e66a0c2985d80aee98a40b0L33-R38) [[4]](https://github.com/Case-Based/casebased/pull/19/files#diff-8695ddc9c0b885aa245ef2d881fdae33ca029cae6e66a0c2985d80aee98a40b0R48-R52)
> * [`tests/test_vocabulary.py`](https://github.com/Case-Based/casebased/pull/19/files#diff-46986df913746988c72be5388d480278d6a98ea3cd44b216ab8d2b4431ad0cd0L15-R15): Fixed the `data_type` parameter in `TargetAttribute` initialization.

Amazing 😍

jonasliendl reviewed (approved):
LGTM, thank you very much!
